src/bindings/python/src/pyopenvino/graph/attribute_visitor.cpp

```diff
@@ -0,0 +1,95 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "pyopenvino/graph/attribute_visitor.hpp"
+
+#include <pybind11/numpy.h>
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+
+#include "openvino/core/attribute_adapter.hpp"
+#include "openvino/core/attribute_visitor.hpp"
+#include "openvino/core/except.hpp"
+#include "pyopenvino/utils/utils.hpp"
+
+namespace py = pybind11;
+
+template <typename AT>
+void visit_attribute(py::dict& attributes,
+                     const std::pair<py::handle, py::handle>& attribute,
+                     ov::AttributeVisitor* visitor) {
+    auto attr_casted = attribute.second.cast<AT>();
+    visitor->on_attribute<AT>(attribute.first.cast<std::string>(), attr_casted);
+    attributes[attribute.first] = std::move(attr_casted);
+
+    return;
+};
+
+void regclass_graph_AttributeVisitor(py::module m) {
+    using PY_TYPE = Common::utils::PY_TYPE;
+
+    py::class_<ov::AttributeVisitor, std::shared_ptr<ov::AttributeVisitor>> visitor(m, "AttributeVisitor");
+
+    visitor.def(
+        "on_attributes",
+        [](ov::AttributeVisitor* self, py::dict& attributes) {
+            py::object float_32_type = py::module_::import("numpy").attr("float32");
+            for (const auto& attribute : attributes) {
+                if (py::isinstance<py::bool_>(attribute.second)) {
+                    visit_attribute<bool>(attributes, attribute, self);
+                } else if (py::isinstance<py::str>(attribute.second)) {
+                    visit_attribute<std::string>(attributes, attribute, self);
+                } else if (py::isinstance<py::int_>(attribute.second)) {
+                    visit_attribute<int64_t>(attributes, attribute, self);
+                } else if (py::isinstance<py::float_>(attribute.second)) {
+                    visit_attribute<double>(attributes, attribute, self);
+                } else if (py::isinstance(attribute.second, float_32_type)) {
+                    visit_attribute<float>(attributes, attribute, self);
+                } else if (py::isinstance<ov::Model>(attribute.second)) {
+                    visit_attribute<std::shared_ptr<ov::Model>>(attributes, attribute, self);
+                } else if (py::isinstance<ov::Dimension>(attribute.second)) {
+                    visit_attribute<ov::Dimension>(attributes, attribute, self);
+                } else if (py::isinstance<ov::PartialShape>(attribute.second)) {
```

Comment:
Should `ov::Shape` be handled too?

Reply:
Actually I looked into serialize `on_adapter` methods and which types there are supported. There is PartialShape, but there isn't Shape.